pkg/operator/controllers/cloudproviderconfig/cloudproviderconfig_controller.go

```diff
@@ -0,0 +1,236 @@
+package cloudproviderconfig
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/base"
+)
+
+const (
+	ControllerName    = "CloudProviderConfig"
+	controllerEnabled = "aro.cloudproviderconfig.enabled"
+)
+
+var cloudProviderConfigName = types.NamespacedName{Name: "cloud-provider-config", Namespace: "openshift-config"}
+
+// / This is a very old version of the CloudProviderConfig struct from k8s upstream
+// / OpenShift diverged from the upstream cluster API and we have a depreciated version of this struct
+// / I was unable to use the recent version of this struct in the upstream k8s Azure Cloud Provider
+// / This version of the struct can be found here:
+// / https://github.com/kubernetes/kubernetes/blob/v1.13.5/pkg/cloudprovider/providers/azure/azure.go#L81
+type azCloudProviderConfig struct {
```

Comment:
It looks like we only use a subset of these fields. Since we're unmarshalling a `[]byte` into this struct, we should be good to omit the fields we don't need ([unmarshal](https://pkg.go.dev/encoding/json#Unmarshal) ignore missing fields).

Reply:
Omitting fields causes JSON marshal errors.